### ISSUE TITLE
Not matching controllers

### ISSUE THREAD

**Issue by erikaxel** [ElMassimo/stimulus-vite-helpers#1]
Hi,

I tried using stimulus-vite-helpers and have run into a problem, it doesn't seem to register my controllers. I have debugged a bit:

First, Im using the following, which is very similar to the code in the README, but with Typescript controllers instead:
```typescript
const application = Application.start()
const controllers = import.meta.globEager('./controllers/**/*_controller.ts')
registerControllers(application, controllers)
```

The reason it then doesn't work for me is that it prefixes the controllers with `controllers--`. 

I digged a bit into the code and found out that if I have a controller in `app/javascript/controllers/dashboard_controller.ts` then `logicalName` in `identifierForGlobKey` is  `controllers/dashboard`. It should have been `dashboard`. 

By changing `CONTROLLER_FILENAME_REGEX` from 
`/^(?:\.\/|.*?(?:controllers|components)\/)?(.+)(?:[_-]controller\..+?)$/` to
`^(?:(?:\.\/|.*)?(?:controllers|components)\/)?(.+)(?:[_-]controller\..+?)$` I got it to work. So properly making the OR part of a group.

It seems like the OR in the regex doesn't work properly, at least in my case.

**Comment by ElMassimo** [comment]
Hi @erikaxel, thanks for reporting! 😃 

As you pointed out, there's a bug in the regex. I added a [test](https://github.com/ElMassimo/stimulus-vite-helpers/blob/main/tests/glob.js) to prevent regressions, which fails in the scenario you described.

The suggested regex causes another common scenario to fail (`./images/reveal_controller.js`), I will tweak it and release a patched version.

**Comment by ElMassimo** [comment]
Fixed in https://github.com/ElMassimo/stimulus-vite-helpers/commit/f2d0c9f22dfed2ea8c3eaee85c7f503a936900c4, and released [`1.0.4`](https://github.com/ElMassimo/stimulus-vite-helpers/blob/main/CHANGELOG.md#stimulus-vite-helpers-104-2021-05-24).

Thanks again for reporting!

**Comment by erikaxel** [comment]
Thanks for the quick turnaround, awesome!
